Userland/Libraries/LibJS/Runtime/AsyncFromSyncIteratorPrototype.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2021, David Tuin <davidot@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "PromiseReaction.h"
+#include <LibJS/Runtime/AsyncFromSyncIteratorPrototype.h>
+#include <LibJS/Runtime/GlobalObject.h>
+#include <LibJS/Runtime/IteratorOperations.h>
+#include <LibJS/Runtime/Promise.h>
+#include <LibJS/Runtime/PromiseConstructor.h>
+
+namespace JS {
+
+AsyncFromSyncIteratorPrototype::AsyncFromSyncIteratorPrototype(GlobalObject& global_object)
+    : PrototypeObject(*global_object.async_iterator_prototype())
+{
+}
+
+void AsyncFromSyncIteratorPrototype::initialize(GlobalObject& global_object)
+{
+    auto& vm = global_object.vm();
+    Object::initialize(global_object);
+
+    u8 attr = Attribute::Writable | Attribute::Configurable;
+    define_native_function(vm.names.next, next, 1, attr);
+    define_native_function(vm.names.return_, return_, 1, attr);
+    define_native_function(vm.names.throw_, throw_, 1, attr);
+}
+
+AsyncFromSyncIteratorPrototype::~AsyncFromSyncIteratorPrototype()
+{
+}
+
+// 27.2.1.1.1 IfAbruptRejectPromise ( value, capability ), https://tc39.es/ecma262/#sec-ifabruptrejectpromise
+#define TRY_OR_REJECT(vm, expression, capability)                                                         \
+    ({                                                                                                    \
+        auto _temporary_result = (expression);                                                            \
+        /* 1. If value is an abrupt completion, then */                                                   \
+        if (_temporary_result.is_error()) {                                                               \
+            vm.clear_exception();                                                                         \
+            vm.stop_unwind();                                                                             \
+                                                                                                          \
+            /* a. Perform ? Call(capability.[[Reject]], undefined, « value.[[Value]] »). */             \
+            (void)vm.call(*capability.reject, js_undefined(), _temporary_result.release_error().value()); \
```

Comment:
Sad clang-format noises but this is what it does

Reply:
I know, we alreadfy have this elsewhere. It's getting confused about the length of `«` and `»` I thikn, but as long as the formatting on CI matches it's ok.